docs/references/architecture/adr-113-mempool-lanes.md

```diff
@@ -294,254 +296,289 @@ TODO
+not support this pattern.
+
+### Custom configuration per lane
+
+A straightforward, future improvement that we leave for after the MVP is to allow customized
+configuration of the lanes instead of sharing the current mempool configuration among lanes. The
+application would need to define new configuration values per lane and pass them to CometBFT during
+the handshake.
+
+### Where to define lanes and priorities
+
+We have considered two alternative approaches for _where_ to configure lanes and priorities:
+1. In `config.toml` or `app.toml`. We have discarded this option as it does not make sense for
+   different nodes to have different lane configurations. The properties defined in the specification above
+   are end-to-end, and so, the lane configuration has to be consistent across the network.
+1. In `ConsensusParams`. There are several disadvantages with this approach. If we allow changing
+   lane information via `ConsensusParams`, the mempool would need to update lanes dynamically. The
+   updating process would be very complex and cumbersome, and not really appealing for an MVP. Two
+   governance proposals would be required to pass to update the lane definitions. A first proposal
+   would be required for upgrading the application, because the lane classification logic (thus the
+   application's code) needs to know the lane configuration beforehand. And a second proposal would
+   be needed for upgrading the lanes via `ConsensusParams`.
+   While it is true that SDK applications could pass a governance proposal with both elements together,
+   it would be something to _always_ do, and it is not clear what the situation would be for non-SDK applications.
+  
+   Also, it is not clear in which order the proposals should apply. The community should be careful
+   not to break performance between the passing of both proposals. The `gov` module could be
+   modified to allow the two changes to be shipped in the same gov proposal, but this does not seem
+   a feasible solution.
+
+Moreover, these two alternatives have a common problem which is how to deal with nodes that are
+late, possibly having lane definitions that do not match with those of nodes at the latest heights.
 
 ## Consequences
 
 TODO
 
 ### Positive
 
+- Application developers will be able to better predict when transactions will be disseminated and
+  processed to be included in a block. This has direct impact on block creation and transaction
+  latency.
+- The mempool will be able to offer Quality of Service guarantees, which does not exist in the
+  current implementation. 
+- Lanes will preserve the "FIFO ordering of transactions" property within the same class (with a
+  best effort approach, as the current implementation).
+- Applications that are unaware of this feature, and therefore not classifying transactions in `CheckTX`
+  will observe the same behavior from the mempool as the current implementation.  
 ### Negative
```

Comment:
On negative, again, are we discussing the starvation problem of low priority lanes. I refer to the `Reap()` part, not the broadcast part, which seems to be implemented using a starvation-free algorithm.

Reply:
The Reap section did not mention the starvation-free property. I corrected it.